### PR TITLE
Test i64 math only in wasm2js mode

### PR DESCRIPTION
**Pull request by kripken**
We could probably also combine these many small programs into multiple functions in a single program. WDYT @sbc100 ?

### CODE REVIEW

File: test/test_core.py [hunk 1]
```diff
@@ -453,60 +453,79 @@ def test_int53_convertI32PairToI53Checked(self):
...
+  @only_wasm2js('test 64-bit math with short values')
   def test_i64_i16(self):
     self.do_core_test('test_i64_i16.c')
 
+  @only_wasm2js('test 64-bit/double conversions')
   def test_i64_qdouble(self):
     self.do_core_test('test_i64_qdouble.c')
 
   def test_i64_varargs(self):
+    # tests va_arg() with i64 params.
```
[the remark this answers is not in the document]
Comment (sbc100) [reply test/test_core.py:528]: I'm pretty sure printf is tested elsewhere/separately.

[the remark this answers is not in the document]
Comment (kripken) [reply test/test_core.py:528]: Most of the tests here use `printf`, but I think keeping one of them is enough? I kept the first.

What I don't know offhand is how `va_arg` works in the wasm backend, and whether we need to test it in this repo - do you?

[the remark this answers is not in the document]
Comment (sbc100) [reply test/test_core.py:528]: va_arg is alwasys a pointer to memory in the wasm backend.   IIUC that is all tested upstream in llvm and defined in tooling conventions.

[the remark this answers is not in the document]
Comment (kripken) [reply test/test_core.py]: Thanks, marked this test as wasm2js-only as well.

File: test/test_core.py [hunk 2]
```diff
@@ -453,59 +453,78 @@ def test_int53_convertI32PairToI53Checked(self):
       self.do_core_test('test_convertI32PairToI53Checked.cpp', interleaved_output=False)
 
   def test_i64(self):
+    # test shifts etc. on 64-bit integers as well as printf() on them. we need
+    # the math testing only for wasm2js but do not apply @only_wasm2js since we
+    # do want some testing of 64-bit printf in our libc (which is not tested in
+    # clang upstream).
```
Comment (sbc100) [inline test/test_core.py:459]: I believe we test printf elsewhere (in fact we might have many overlapping printf tests).  e.g. test/printf/test.c

**Review by sbc100 — APPROVED** [review]
lgtm % comment

File: test/test_core.py (hunk 2 above)
Comment (kripken) [reply test/test_core.py]: Good point, removed this one too.

**Pull request merged**